packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -445,21 +454,74 @@ const DataTable = ({
     ],
   );
 
-  const dataTableFilters = useMemo(
-    () =>
-      columns
-        .filter((column) => column.enableColumnFilter !== false)
-        .map((column) => {
-          return {
-            id: column.accessorKey as string,
+  /**
+   * This hack is to provide compatibility with Material-React-Table's
+   * filterOptions format, which allows for strings and { label: string, value: string }
+   */
+  const convertFilterSelectOptions = useCallback(
+    (options: DataTableColumn<DataTableRowData>["filterSelectOptions"]) =>
+      options?.map((option) =>
+        typeof option === "string"
+          ? {
+              label: option,
+              value: option,
+            }
+          : {
+              // If the option isn't a string, it must have value and/or option defined
+              // If either is undefined, use the other
+              label: option.label ?? option.value,
+              value: option.value ?? option.label,
+            },
+      ),
+    [],
+  );
+
+  const convertColumnToFilter = useCallback(
+    (column: DataTableColumn<DataTableRowData>) =>
+      column.enableColumnFilter && column.accessorKey
+        ? ({
+            id: column.accessorKey,
             label: column.header,
-            variant: column.filterVariant ?? "text",
-            options: column.filterSelectOptions,
-          } as DataFilter;
-        }),
-    [columns],
+            variant: column.filterVariant,
+            options: convertFilterSelectOptions(column.filterSelectOptions),
+          } satisfies DataFilter as DataFilter)
+        : null,
+    [],
   );
 
+  /**
+   * Filters default to the columns, but can be overridden
+   * with the `filters` prop. `filters` should be an array
+   * of column accessorKeys, column defs, or DataFilters.
+   */
+  const dataTableFilters = useMemo(() => {
+    const providedFilters = filtersProp || columns;
+    return providedFilters.reduce<DataFilter[]>((accumulator, item) => {
+      if (typeof item === "string") {
+        const foundColumn = columns.find(
+          (column) => column.accessorKey === item,
+        );
+        if (foundColumn) {
+          const filter = convertColumnToFilter(foundColumn);
+          if (filter) {
+            accumulator.push(filter);
+          }
+        }
+      } else if ("accessorKey" in item) {
+        // Checks if it's a column
+        const filter = convertColumnToFilter(item);
+        if (filter) {
+          accumulator.push();
+        }
+      } else if ("label" in item) {
+        // Checks if it's a DataFilter
+        accumulator.push(item);
+      }
+      // If none of the conditions match, item is ignored (not mapping to undefined)
+      return accumulator;
+    }, []);
+  }, [columns, filtersProp]);
```

Comment:
We should rewrite this to note use `.push`. That's an array mutation, and we don't (or shouldn't) have that pattern anywhere in our codebase.

Mutations lead to hard-to-fix bugs and make it difficult to debug because you're taking an object reference and changing it on the fly.

Instead, we should be creating new arrays with `.concat`.

In this case, the whole function could easily be converted to using `.concat` with more `return` statements:
```suggestion
const dataTableFilters = useMemo(() => {
  const providedFilters = filtersProp || columns;
  return providedFilters.reduce<DataFilter[]>((accumulator, item) => {
    if (typeof item === "string") {
      const foundColumn = columns.find(
        (column) => column.accessorKey === item,
      );
      if (foundColumn) {
        const filter = convertColumnToFilter(foundColumn);
        if (filter) {
          return accumulator.concat(filter);
        }
      }
    } else if ("accessorKey" in item) {
      // Checks if it's a column
      const filter = convertColumnToFilter(item);
      if (filter) {
        return accumulator.concat();
      }
    } else if ("label" in item) {
      // Checks if it's a DataFilter
      return accumulator.concat(item);
    }
    // If none of the conditions match, item is ignored (not mapping to undefined)
    return accumulator;
  }, []);
}, [columns, filtersProp]);
```